### PR TITLE
#29 Replace http dogu registry implementation

### LINKED ISSUE #29: Use http dogu registry from cesapp-lib

### PR DESCRIPTION
**Pull request by Subaro**
We implemented a simple HTTP dogu registry at the beginning of the development. This PR replaces the simple implementation with the one from the `cesapp-lib`. 

Resolves #29

### CODE REVIEW

File: controllers/config/config_test.go [hunk 1]
```diff
@@ -16,13 +15,13 @@ func TestNewOperatorConfig(t *testing.T) {
 	_ = os.Unsetenv("DOCKER_REGISTRY")
 
 	expectedNamespace := "myNamepsace"
-	expectedDoguRegistryData := config.DoguRegistryData{
+	expectedDoguRegistryData := DoguRegistryData{
 		Endpoint: "myEndpoint",
 		Username: "myUsername",
 		Password: "myPassword",
 	}
 	inputDockerRegistrySecretData := "{\"auths\":{\"your.private.registry.example.com\":{\"username\":\"myDockerUsername\",\"password\":\"myDockerPassword\",\"email\":\"jdoe@example.com\",\"auth\":\"c3R...zE2\"}}}"
```
Comment (ppxl) [inline controllers/config/config_test.go:23]: Tests are a sort of documentation. Triple backticks ` ``` "key"="value" ``` `can drastically reduce the mental load of reading this JSON text. I often find myself looking into tests to scout for examples or to grasp finer details of configuration.

**Review by ppxl — CHANGES_REQUESTED** [review]
Good work. There are some small things that we should look into but nothing serious only superficial stuff.

File: controllers/config/config_test.go (hunk 1 above)
Comment (Subaro) [reply controllers/config/config_test.go:23]: I think you meant the single backticks 
``` 
`"key"="value"`
```
Done.

File: controllers/dogu_manager.go [hunk 2]
```diff
@@ -116,7 +112,11 @@ type applier interface {
 
 // NewDoguManager creates a new instance of DoguManager
 func NewDoguManager(client client.Client, operatorConfig *config.OperatorConfig, cesRegistry cesregistry.Registry) (*DoguManager, error) {
-	doguRemoteRegistry := registry.New(operatorConfig.DoguRegistry.Username, operatorConfig.DoguRegistry.Password, operatorConfig.DoguRegistry.Endpoint)
+	doguRemoteRegistry, err := cesremote.New(operatorConfig.GetRemoteConfiguration(), operatorConfig.GetRemoteCredentials())
+	if err != nil {
+		return nil, fmt.Errorf("failed find create new remote dogu registry: %w", err)
```
[the remark this answers is not in the document]
Comment (Subaro) [reply controllers/dogu_manager.go:117]: Done.

File: controllers/suite_int_test.go [hunk 3]
```diff
@@ -59,6 +62,13 @@ func TestAPIs(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
+	// we need to ensure that the development stage flag is not passed by our makefiles
```
[the remark this answers is not in the document]
Comment (Subaro) [reply controllers/suite_int_test.go:65]: The negative consequences are that we test our operator in the development stage mode. This mode changes some behavior of the operator itself. I will add some more information to the comment.

File: controllers/config/config_test.go [hunk 4]
```diff
@@ -103,11 +102,58 @@ func TestNewOperatorConfig(t *testing.T) {
...
 		assert.Nil(t, operatorConfig)
 	})
 }
+
+func TestOperatorConfig_GetRemoteConfiguration(t *testing.T) {
+	tests := []struct {
+		name           string
+		inputEndpopint string
+	}{
+		{name: "get remote configuration with correct url", inputEndpopint: "https://dogu.cloudogu.com/api/v2/"},
```
[the remark this answers is not in the document]
Comment (Subaro) [reply controllers/config/config_test.go:119]: Thanks :)

File: controllers/suite_int_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Subaro) [reply controllers/suite_int_test.go]: Done

**Comment by ppxl** [comment]
LGTM

**Pull request merged**